astropy/time/formats.py

```diff
@@ -408,9 +409,13 @@ class TimeNumeric(TimeFormat):
 
     def _check_val_type(self, val1, val2):
         """Input value validation, typically overridden by derived classes"""
+        # Save original state of val2 because the super()._check_val_type below
+        # may change val2 from None to np.array(0).
+        val2_is_none = val2 is None
+
         if val1.dtype.kind == 'f':
             val1, val2 = super()._check_val_type(val1, val2)
```

Comment:
Looks like `val2` is overwritten here but `val2_is_none` was set based on the older value. Is this a bug?

Reply:
No, it is the point in this case, though that you ask suggests the comment above was not clear enough.
@taldcroft - since this is the only place where `val2` can get messed up, I would suggest fixing it here too. Maybe simply
```
if val2 is None:
    val1, _  = super()._check_val_type(val1, None)
else:
    val1, val2 = super()._check_val_type(val1, val2)
```
or
```
val1, new_val2 = super()._check_val_type(val1, val2)
if val2 is not None:
    val2 = new_val2
```
(though possibly it should be solved in the routine above, perhaps with a new argument - bit out of scope, though)

Reply:
The `elif` is not affected by this, which is why I was suggesting to keep it local, but really this is hairsplitting...